integrations/azure_ai_search/src/haystack_integrations/document_stores/azure_ai_search/document_store.py

```diff
@@ -0,0 +1,450 @@
+        """
+        Returns how many documents are present in the search index.
+
+        :param kwargs: additional keyword parameters.
+        :returns: list of retrieved documents.
+        """
+        return self.client.get_document_count(**kwargs)
+
+    def write_documents(self, documents: List[Document], policy: DuplicatePolicy = DuplicatePolicy.FAIL) -> int:
+        """
+        Writes the provided documents to search index.
+
+        :param documents: documents to write to the index.
+        :return: the number of documents added to index.
+        """
+
+        def _convert_input_document(documents: Document):
+            document_dict = asdict(documents)
+            if not isinstance(document_dict["id"], str):
+                msg = f"Document id {document_dict['id']} is not a string, "
+                raise Exception(msg)
+            index_document = self._convert_haystack_documents_to_azure(document_dict)
+
+            return index_document
+
+        if len(documents) > 0:
+            if not isinstance(documents[0], Document):
+                msg = "param 'documents' must contain a list of objects of type Document"
+                raise ValueError(msg)
+
+        documents_to_write = []
+        for doc in documents:
+            try:
+                self.client.get_document(doc.id)
+                if policy == DuplicatePolicy.SKIP:
+                    logger.info(f"Document with ID {doc.id} already exists. Skipping.")
+                    continue
+                elif policy == DuplicatePolicy.FAIL:
+                    msg = f"Document with ID {doc.id} already exists."
+                    raise DuplicateDocumentError(msg)
+                elif policy == DuplicatePolicy.OVERWRITE:
+                    logger.info(f"Document with ID {doc.id} already exists. Overwriting.")
+                    documents_to_write.append(_convert_input_document(doc))
+            except ResourceNotFoundError:
+                # Document does not exist, safe to add
+                documents_to_write.append(_convert_input_document(doc))
+
+        if documents_to_write != []:
+            self.client.upload_documents(documents_to_write)
+        return len(documents_to_write)
+
+    def delete_documents(self, document_ids: List[str]) -> None:
+        """
+        Deletes all documents with a matching document_ids from the search index.
+
+        :param document_ids: ids of the documents to be deleted.
+        """
+        if self.count_documents == 0:
+            return
+        documents = self._get_raw_documents_by_id(document_ids)
+        if documents:
+            self.client.delete_documents(documents)
+
+    def get_documents_by_id(self, document_ids: List[str]) -> List[Document]:
+        return self._convert_search_result_to_documents(self._get_raw_documents_by_id(document_ids))
+
+    def search_documents(self, search_text: Optional[str] = "*", top_k: Optional[int] = 10) -> List[Document]:
```

Comment:
We generally don't expose such methods directly but through a retriever. Is that different?

`top_k: Optional[int] = 10` -> `top_k: int = 10`
same for `search_text`

Reply:
Found a similar method in Astra, so I thought we can keep it.
https://github.com/deepset-ai/haystack-core-integrations/blob/e97cfee4c2b0868d96e4a4a335ec5cd7ee0f6cf1/integrations/astra/src/haystack_integrations/document_stores/astra/document_store.py#L354

Reply:
Sorry I was talking fof `search_documents`

Reply:
In that case, would you prefer to have `search_documents` as an internal method used for content filtering. This can be called by user through `filter_documents` using `content` key?